mlflow/deployments/databricks/__init__.py

```diff
@@ -207,6 +245,79 @@ def predict(self, deployment_name=None, inputs=None, endpoint=None):
             timeout=MLFLOW_DEPLOYMENT_PREDICT_TIMEOUT.get(),
         )
 
+    @experimental
+    def predict_stream(
+        self, deployment_name=None, inputs=None, endpoint=None
+    ) -> Iterator[Dict[str, Any]]:
+        """
+        Submit a query to a configured provider endpoint, and get streaming response
+
+        Args:
+            deployment_name: Unused.
+            inputs: The inputs to the query, as a dictionary.
+            endpoint: The name of the endpoint to query.
+
+        Returns:
+            An iterator of dictionary containing the response from the endpoint.
+
+        Example:
+
+        .. code-block:: python
+
+            from mlflow.deployments import get_deploy_client
+
+            client = get_deploy_client("databricks")
+            chunk_iter = client.predict_stream(
+                endpoint="databricks-llama-2-70b-chat",
+                inputs={
+                    "messages": [{"role": "user", "content": "Hello!"}],
+                    "temperature": 0.0,
+                    "n": 1,
+                    "max_tokens": 500,
+                },
+            )
+            for chunk in chunk_iter:
+                print(chunk)
+                # Example:
+                # {
+                #     "id": "82a834f5-089d-4fc0-ad6c-db5c7d6a6129",
+                #     "object": "chat.completion.chunk",
+                #     "created": 1712133837,
+                #     "model": "llama-2-70b-chat-030424",
+                #     "choices": [
+                #         {
+                #             "index": 0, "delta": {"role": "assistant", "content": "Hello"},
+                #             "finish_reason": None,
+                #         }
+                #     ],
+                #     "usage": {"prompt_tokens": 11, "completion_tokens": 1, "total_tokens": 12},
+                # }
+        """
+        inputs = inputs or {}
+
+        # Add stream=True param in request body to get streaming response
+        # See https://docs.databricks.com/api/workspace/servingendpoints/query#stream
+        chunk_line_iter = self._call_endpoint_stream(
+            method="POST",
+            prefix="/",
+            route=posixpath.join(endpoint, "invocations"),
+            json_body={**inputs, "stream": True},
+            timeout=MLFLOW_DEPLOYMENT_PREDICT_TIMEOUT.get(),
+        )
+
+        for line in chunk_line_iter:
```

Comment:
any chance that `line` is a space?

Reply:
Added checking:
```
            splits = line.split(":", 1)
            if len(splits) < 2:
                raise MlflowException(f"Unknown streaming response format: '{line}'.")
```

Reply:
can we do `line.startswith('data:')`?

Reply:
I think `line.split(":", 1)` should be more robust,

becase 'data : value' might be also a valid case, but `line.startswith('data:')` fails in this case

Reply:
got it